### ISSUE TITLE
usage of `assertthat::assert_that` for input argument checks

### ISSUE THREAD

**Issue by stijnvanhoey** [adokter/bioRad#97]
Currently, Biorad uses `if (...) {stop(...)}` (with custom error message) and `stopifnot(...)` (without custom message) constructions to do input function validation. 

Making one-liners of the `if (...) {stop(...)}` construction is resulting in [long lines](https://github.com/adokter/bioRad/blob/rename_functions/R/nexrad_odim.R#L22) and making it all spread on multiple lines (cfr. syntax formatting) makes the functions _long_. 

Could we rewrite all of the input checks by using the `assertthat::assert_that(condition, msg =  "...")` functionality? This would give all input checks a similar result and get rid of all these short `if`-constructions.

**Comment by adokter** [comment]
Yes I learnt of the assertthat package recently, and wished I had used that from the start. +1 for changing that (now, or in the future)

**Comment by stijnvanhoey** [comment]
Check! I'm not taking it into account in the current renaming work, but let's put this on the enahncement list!

**Comment by stijnvanhoey** [comment]
To have a safe roadmap and at the same time gain test-coverage: ;-)

1. create tests for the current input checks, make them work
2. rewrite with `assertthat`

**Comment by adokter** [comment]
Issue #148 requires unit testing, therefore rewrite with `assertthat` likely part of rOpenSci onboarding

**Comment by PietrH** [comment]
I had a quick look at how much work this would be, i think the `stopifnot()` assertations can pretty easily be replaced, we'll probably need to be a bit more careful with `stop()`. Writing tests first seems like the sensible thing to do. 

```r
#checking where we could replace assertations with assertthat::assert_that()
library(dplyr)

stopifnot_mentions <- system('grep "stopifnot(" -r .', intern = TRUE)
stop_mentions <- system('grep "stop(" -r . --include \\*.R', intern = TRUE)

count_mentions <- function(mentions) {
  
  readr::read_delim(
    paste(mentions, collapse = "\n"),
    col_names = c("file","mention")) %>%
    count(file,sort = TRUE) %>%
    print(n = Inf)
}

# should be easy replacements
count_mentions(stopifnot_mentions)
# need more review
count_mentions(stop_mentions)
```

For `stopifnot()`:
|file                                                              |  n|
|:-----------------------------------------------------------------|--:|
|./R/integrate_profile.R                                           | 10|
|./R/doy_noy.R                                                     |  8|
|./R/rcs.R                                                         |  6|
|./R/vpts.R                                                        |  5|
|./R/check_night.R                                                 |  4|
|./R/sd_vvp_threshold.R                                            |  4|
|./R/select.R                                                      |  4|
|./R/get_elevation_angles.R                                        |  3|
|./R/plot.vpts.R                                                   |  3|
|./R/ppi.R                                                         |  3|
|./R/project_as_ppi.R                                              |  3|
|./R/scan_convert.R                                                |  3|
|./R/as.data.frame.R                                               |  2|
|./R/get_quantity.R                                                |  2|
|./R/plot.vp.R                                                     |  2|
|./R/pvol.R                                                        |  2|
|./R/scan.R                                                        |  2|
|./R/vp.R                                                          |  2|
|./.Rproj.user/317CC55D/sources/session-9746c1c8/427B6130-contents |  1|
|./.Rproj.user/317CC55D/sources/session-9746c1c8/B9DCB993-contents |  1|
|./R/bind_into_vpts.R                                              |  1|
|./R/download_basemap.R                                            |  1|
|./R/map.R                                                         |  1|
|./R/param.R                                                       |  1|
|./R/plot.ppi.R                                                    |  1|
|./R/plot.scan.R                                                   |  1|
|./R/plot.vpi.R                                                    |  1|
|./R/vpi.R                                                         |  1|

For `stop()`:
|file                    |  n|
|:-----------------------|--:|
|./R/integrate_to_ppi.R  | 24|
|./R/scan_convert.R      | 21|
|./R/bind_into_vpts.R    | 13|
|./R/read_pvolfile.R     | 11|
|./R/beam.R              |  9|
|./R/composite_ppi.R     |  9|
|./R/plot.vpts.R         |  9|
|./R/read_vpts.R         |  9|
|./R/map.R               |  7|
|./R/operators.R         |  7|
|./R/apply_mistnet.R     |  5|
|./R/calculate_vp.R      |  4|
|./R/download_vpfiles.R  |  3|
|./R/integrate_profile.R |  3|
|./R/read_cajun.R        |  3|
|./R/select.R            |  3|
|./R/as.data.frame.R     |  2|
|./R/download_basemap.R  |  2|
|./R/plot.ppi.R          |  2|
|./R/plot.scan.R         |  2|
|./R/plot.vp.R           |  2|
|./R/plot.vpi.R          |  2|
|./R/rcs.R               |  2|
|./R/regularize_vpts.R   |  2|
|./R/sd_vvp_threshold.R  |  2|
|./R/check_night.R       |  1|
|./R/color_scale.R       |  1|
|./R/get_param.R         |  1|
|./R/get_quantity.R      |  1|
|./R/read_vpfiles.R      |  1|

**Comment by PietrH** [comment]
I've started a [branch](https://github.com/adokter/bioRad/tree/improve-error-test-coverage) where we can improve code coverage for the messages first, as some are untested. In a second PR/branch we can then replace the current error messaging with `assertthat::assert_that()`. We could also write custom error messages in a third stage.

**Comment by peterdesmet** [comment]
_Discussed at coding sprint in Amsterdam_

We want to adopt {cli} in bioRad, which doesn't work super well with {assertthat}, so I wouldn't pursue this further
